examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -0,0 +1,298 @@
+        0x7c,
+    };
+
+    return CopySpanToMutableSpan(ByteSpan{ kCdForAllExamples }, out_cd_buffer);
+
+#else
+    size_t buflen = out_cd_buffer.size();
+    ChipLogDetail(Crypto, "Get certificate declaration from se05x");
+    ReturnErrorOnFailure(se05xGetCertificate(CD_CERT_DECLARATION_DATA_SE05X_ID, out_cd_buffer.data(), &buflen));
+    out_cd_buffer.reduce_size(buflen);
+    return CHIP_NO_ERROR;
+#endif
+}
+
+CHIP_ERROR ExampleSe05xDACProviderv2::GetFirmwareInformation(MutableByteSpan & out_firmware_info_buffer)
+{
+    // TODO: We need a real example FirmwareInformation to be populated.
+    out_firmware_info_buffer.reduce_size(0);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ExampleSe05xDACProviderv2::SignWithDeviceAttestationKey(const ByteSpan & message_to_sign,
+                                                                   MutableByteSpan & out_signature_buffer)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrReturnError(IsSpanUsable(out_signature_buffer), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(IsSpanUsable(message_to_sign), CHIP_ERROR_INVALID_ARGUMENT);
+
+    ChipLogDetail(Crypto, "Sign using DA key from se05x (Using internal sign)");
+
+    TLV::TLVReader msg_reader;
+    TLV::TLVReader tagReader;
+
+    msg_reader.Init(message_to_sign);
+
+    /* To be removed. Use common key id to sign message */
+    static bool sign_cert_decl_attest = 1;
+
+    if (sign_cert_decl_attest)
+    {
+        /* Check if certificate declaration tag is present and Skip certificate declaration tag */
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
+        uint8_t attlen = tagReader.GetLength();
+        VerifyOrReturnError(attlen > 0, CHIP_ERROR_INVALID_TLV_TAG);
+        /* Get attestation nonce */
+        ByteSpan attest_nonce;
+        ReturnErrorOnFailure(tagReader.Get(attest_nonce));
+        /* Set attestation nonce */
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(3), tagReader));
+        uint8_t tslen = tagReader.GetLength();
+        if (tslen > 0)
+        {
+            ByteSpan time_stamp;
+            ReturnErrorOnFailure(tagReader.Get(time_stamp));
+            /* Set time stamp data */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_TIME_STAMP_DATA_SE05X_ID, time_stamp.data(), time_stamp.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+        /* Set time stamp length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tslen, 1), CHIP_ERROR_INTERNAL);
```

Comment:
Why isn't this using `time_stamp.size()`?  That is, what is the point of `tslen` at all?  It seems like an attractive nuisance, as do most of the `GetLength` calls here.

Reply:
get() member will return error in case the tag length is 0. So using the GetLength() before get() member is called.

Reply:
Wait, what?  Why would `Get` return an error on 0 length?  It most certainly does not do that...

Reply:
I have created the issue to check this as well as other changes you suggested in the previous conversation. (https://github.com/project-chip/connectedhomeip/issues/23064). Will address these in the next PR.